kpi/tests/api/v1/test_api_submissions.py

```diff
@@ -66,6 +71,52 @@ def test_list_submissions_limit(self):
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response.data), limit)
 
+    def test_delete_submission_owner(self):
+        submission = self.submissions[0]
+        url = self.asset.deployment.get_submission_detail_url(submission.get(
+            self.asset.deployment.INSTANCE_ID_FIELDNAME))
+
+        response = self.client.delete(url,
+                                      content_type="application/json",
+                                      HTTP_ACCEPT="application/json")
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        response = self.client.get(self.submission_url, {'format': 'json'})
```

Comment:
The good way should to pass the format, but it calls `get_submission_detail_url()` which hardcodes the url without any extension. Because of that, DRF returns the default renderer which is the Html.  It always returns a `200` instead of a `204` (for `DELETE`) 
HTTP_ACCEPT="application/json"  forces DRF to return the JSON response. 

`content_type` is useless in that case.

Reply:
We did a little testing and saw that sadly the "right" way does depend upon the method. The `get()` method has a magical translation from its `data` parameter, to which we might pass `{'format': 'json'}`, to the `QUERY_STRING` HTTP header:
https://github.com/encode/django-rest-framework/blob/98e56e0327596db352b35fa3b3dc8355dc9bd030/rest_framework/test.py#L194-L197

No such magic exists for `delete()`, leaving us the options:
1. Change the URL to end in `.json`
1. Pass `HTTP_ACCEPT="application/json"` to `delete()` (this does work :+1:)
1. Pass `QUERY_STRING='format=json'` to `delete()`

`HTTP_ACCEPT` (as it is) is fine; let's remove `content_type` if it's useless.

As an alternative to modifying `get_submission_detail_url()`—and generally to worrying about whether or not a URL-generating function returns a trailing slash—we could use `reverse()` and your `_get_endpoint()` method to construct the submission detail URL

Reply:
Done in https://github.com/kobotoolbox/kpi/pull/2900/commits/3385ea9379a30bc0a139ded3220d1a09c13e93b6